### PR TITLE
Website frontpage

### PR DESCRIPTION
**Pull request by MasseGuillaume**
2018-08-24-172632_1243x1089_scrot

### CODE REVIEW

File: website/i18n/en.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "localized-strings": {
     "next": "Next",
     "previous": "Previous",
-    "tagline": "Refactoring and linting tool for Scala",
```
Comment (olafurpg) [inline website/i18n/en.json]: Let's keep this tagline since this is what appears as title from Google search.

File: website/core/Footer.js [hunk 2]
```diff
@@ -1,73 +1,8 @@
...
-            <h5>More</h5>
-            <a href={siteConfig.repoUrl} target="_blank">
-              GitHub
-            </a>
-          </div>
-        </section>
-        <section className="copyright">{copyright}</section>
-      </footer>
-    );
+    return null
```
Comment (olafurpg) [inline website/core/Footer.js]: Can we please keep the footer? I find it helpful to have a lot of links at the bottom when navigating a site.

File: scalafix-docs/src/main/scala/docs/website.scala [hunk 3]
```diff
@@ -23,32 +23,30 @@ package object website {
...
-      tbody(rows)
-    )
-    html.toString()
+      }
+    }
+
+    div(
+      h2("Semantic Rules"),
+      buildTable(semantic),
+      h2("Syntaxtic Rules"),
```
Comment (olafurpg) [inline scalafix-docs/src/main/scala/docs/website.scala]: Syntactic

File: website/pages/en/index.js [hunk 4]
```diff
@@ -58,35 +62,21 @@ const Logo = props => (
...
   render() {
     let language = this.props.language || "";
     return (
       <SplashContainer>
-        {/* <Logo img_src={imgUrl('scalameta-logo.png')} /> */}
         <div className="inner">
           <ProjectTitle />
-          <PromoSection>
-            <Button href={docUrl("users/installation.html", language)}>
-              Documentation
```
Comment (olafurpg) [inline website/pages/en/index.js]: We should keep a "call to action" button. Currently, it's not clear that you should click on the front page.

File: website/pages/en/index.js [hunk 5]
```diff
@@ -58,35 +62,21 @@ const Logo = props => (
 
 const ProjectTitle = props => (
   <h2 className="projectTitle">
-    {siteConfig.title}
-    <small>{siteConfig.tagline}</small>
```
Comment (olafurpg) [inline website/pages/en/index.js]: I preferred the older ProjectTitle, the font change and logo are too much IMO

screen shot 2018-08-24 at 17 50 58

I don't think it's necessary to customize this.

File: scalafix-docs/src/main/scala/docs/website.scala [hunk 6]
```diff
@@ -23,32 +23,30 @@ package object website {
...
-        td(semantic),
-        td(ruleLink(rule.name.value)),
-        td(rule.description)
-      )
-    }
-    val html = table(
-      thead(
+    val (semantic, syntaxtic) = rules.partition(_.isInstanceOf[v1.SemanticRule])
+
+    def buildTable(rules: List[v1.Rule]) = {
```
Comment (olafurpg) [inline scalafix-docs/src/main/scala/docs/website.scala]: This method returns Unit, to fix it
```diff
diff --git a/scalafix-docs/src/main/scala/docs/website.scala b/scalafix-docs/src/main/scala/docs/website.scala
index c00761de..aca497dc 100644
--- a/scalafix-docs/src/main/scala/docs/website.scala
+++ b/scalafix-docs/src/main/scala/docs/website.scala
@@ -23,10 +23,11 @@ package object website {
       .filterNot(_.name.isDeprecated)
       .sortBy(_.name.value)
 
+    pprint.log(rules.map(_.name.value))
     val (semantic, syntaxtic) = rules.partition(_.isInstanceOf[v1.SemanticRule])
 
-    def buildTable(rules: List[v1.Rule]) = {
-      val rows: List[Text.TypedTag[String]] = rules.map { rule =>
+    def buildTable(rules: List[v1.Rule]): List[Text.TypedTag[String]] = {
+      rules.map { rule =>
         val docPath = PathIO.workingDirectory
           .resolve("docs")
           .resolve("rules")

```

After that, it renders weird

screen shot 2018-08-24 at 17 54 53

File: website/pages/en/index.js [hunk 7]
```diff
@@ -103,52 +93,131 @@ const Block = props => (
...
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+class Highlight extends React.Component {
+  render () {
+    let content = {
```
Comment (gabro) [inline website/pages/en/index.js]: not a big deal, but I would replace all `let` with `const`

File: website/pages/en/index.js [hunk 8]
```diff
@@ -103,52 +107,132 @@ const Block = props => (
...
-      style={{ textAlign: "center" }}
-    >
-      {features.map(feature => (
-        <Block key={feature.title}>{[feature]}</Block>
-      ))}
-    </div>
-  );
-};
+    return (
+      <pre className="{this.props.lang} hljs">
```
Comment (gabro) [inline website/pages/en/index.js]: This is a mistake, `this.props.lang` is not getting interpolated here.

Fix:

```
      <pre className={`${this.props.lang} hljs`}>
```

**Pull request merged**